### ISSUE TITLE
Creating files in /system/vendor prevents boot

### ISSUE THREAD

**Issue by skalnet** [topjohnwu/Magisk#10]
I'm trying to create a module for the updated adreno 330 beta drivers for my Samsung Galaxy S5 (g900f).
But after installing the module, the device won't boot (not even Boot animation will show).

There are two files, that don't exist in the original system:

```
/system/vendor/firmware/a330_pfp.fw
/system/vendor/firmware/a330_pm4.fw
```

Leaving these two files out of the module results in the device booting up normally.
Creating files with these names in system and the device also boots up normally.
Seems as adding new files or folders to /system/vendor doesn't work the way it should, as two users at XDA experienced something similar

http://forum.xda-developers.com/showpost.php?p=69261346&postcount=50
http://forum.xda-developers.com/showpost.php?p=69261346&postcount=51

**Comment by topjohnwu** [comment]
Sorry missed your issue (been busy lately)
I think it is caused by the folder "firmware", as my V4A module also patches /system/vendor (/vendor).
I'll try to see what happen when I add files to /system/vendor/firmware.
Thanks for reporting!!

**Comment by jsaxon2** [comment]
I have the same type of problem if I add my Magisk Mod for Substranium overlays. If I create the /system/vender/overlay directory, the mod works fine, but if the overlay directory is not present in the actual system/vendor, phone just reboots into recovery with the mod enabled. This used to work on Magisk v6 and started happening with version 7 and 8.

**Comment by topjohnwu** [comment]
@jsaxon2, I also have an HTC 10, could you directly send me a mod that prevents boot?
That will make the debug process much easier

**Comment by jsaxon2** [comment]
This just adds a layer for AOSP type signal bars. If I flash, phone reboots into recovery. If I then manually create an empty overlay directory in /system/vendor/ and phone boots normal with layer applied.

https://drive.google.com/open?id=0B1iRsZPFZo3ESEl0am95eTNNMVU

**Comment by jsaxon2** [comment]
This just adds a layer for AOSP type signal bars. If I flash, phone reboots
into recovery. If I then manually create an empty overlay directory in
/system/vendor/ and phone boots normal with layer applied.

https://drive.google.com/open?id=0B1iRsZPFZo3ESEl0am95eTNNMVU

**Comment by topjohnwu** [comment]
Fixed in f1d3e35aacdaf8bc0e56246129805077f3431b9d

**Comment by laggardkernel** [comment]
@topjohnwu Still some problems concerning mounting files under /system/vendor. Incomplete bind_mount happened on Magisk v9 under /system/vendor/. 

Modified audio confs, /magisk/v4a/system/vendor/etc/audio_output_policy.conf and /magisk/v4a/system/vendor/etc/audio_effects.conf, were ignored to be mounted under /system/vendor  when overlay theme apks were mounted. See the Magisk log below,
```
** Magisk post-fs mode running...
** Magisk post-fs-data mode running...
Preparing modules
Bind mount dummy system
Mount: /dev/magisk/dummy/system/lib/soundfx
Mount: /dev/magisk/dummy/system/fonts
Mount: /dev/magisk/dummy/system/vendor
Mount: /dev/magisk/dummy/system/app
Mount: /dev/magisk/dummy/system/priv-app
Bind mount module items
Mount: /magisk/v4a/system/lib/soundfx/libv4a_fx_ics.so
Mount: /magisk/overlay/system/vendor/overlay
Mount: /magisk/ubuntu_font/system/fonts/Ubuntu-RI.ttf
Mount: /magisk/ubuntu_font/system/fonts/Ubuntu-R.ttf
```
But after I created a empty overlay folder under /system/vendor, the modified audio confs are mounted correctly. /system/vendor/overlay does NOT exist in the OxygenOS ROM originally.
```
** Magisk post-fs mode running...
** Magisk post-fs-data mode running...
Preparing modules
Bind mount dummy system
Mount: /dev/magisk/dummy/system/lib/soundfx
Mount: /dev/magisk/dummy/system/fonts
Mount: /dev/magisk/dummy/system/app
Mount: /dev/magisk/dummy/system/priv-app
Bind mount module items
Mount: /magisk/v4a/system/lib/soundfx/libv4a_fx_ics.so
Mount: /magisk/v4a/system/vendor/etc/audio_output_policy.conf
Mount: /magisk/v4a/system/vendor/etc/audio_effects.conf
Mount: /magisk/overlay/system/vendor/overlay
Mount: /magisk/ubuntu_font/system/fonts/Ubuntu-RI.ttf
Mount: /magisk/ubuntu_font/system/fonts/Ubuntu-R.ttf
```

OnePlus3, OxygenOS 3.2.7, Marshmallow 6.0.1.
